### PR TITLE
feat: Enable tls for sidetree-mock server

### PR DESCRIPTION
**Pull request by fqutishat**
Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: scripts/generate_test_keys.sh [hunk 1]
```diff
@@ -8,20 +8,27 @@
 set -e
 
 
-echo -e  "[SAN]\nsubjectAltName=DNS:*.example.com,DNS:localhost" >> /etc/ssl/openssl.cnf
-
 echo "Generating sidetree-mock Test PKI"
-cd /opt/go/src/github.com/trustbloc/sidetree-mock
+
+cd /opt/workspace/sidetree-mock
```
Comment (troyronda) [inline scripts/generate_test_keys.sh]: you should re-use the sandbox CA script rather than creating a new one.

Comment (troyronda) [reply scripts/generate_test_keys.sh]: See https://github.com/trustbloc/edge-sandbox/blob/master/scripts/generate_test_keys.sh#L34

Comment (fqutishat) [reply scripts/generate_test_keys.sh]: this is just for bdd-test

Comment (troyronda) [reply scripts/generate_test_keys.sh]: and https://github.com/trustbloc/edge-sandbox/blob/master/Makefile#L120

File: test/bddtests/restclient/sender.go [hunk 2]
```diff
@@ -30,7 +31,12 @@ func SendRequest(url string, req []byte) (*HttpRespone, error) {
 
 // SendResolveRequest send a regular GET request to the sidetree-mock and expects 'side tree document' argument as a response
 func SendResolveRequest(url string) (*HttpRespone, error) {
-	client := &http.Client{}
+	client := &http.Client{
+		// TODO add tls config
+		// TODO !!!!!!!remove InsecureSkipVerify after configure tls for http client
```
Comment (troyronda) [inline test/bddtests/restclient/sender.go]: Yup need to fix these.

Comment (rolsonquadras) [inline test/bddtests/restclient/sender.go]: May be we can add a common trustbloc issue # to this TODO. It would be easy to search all with that issue.

Comment (fqutishat) [reply test/bddtests/restclient/sender.go]: how

Comment (fqutishat) [reply test/bddtests/restclient/sender.go]: we need to add it per project

File: scripts/generate_test_keys.sh (hunk 1 above)
Comment (fqutishat) [reply scripts/generate_test_keys.sh]: I add TODO

File: test/bddtests/restclient/sender.go (hunk 2 above)
Comment (fqutishat) [reply test/bddtests/restclient/sender.go]: I add TODO

Comment (fqutishat) [reply test/bddtests/restclient/sender.go]: I add TODO

Comment (rolsonquadras) [reply test/bddtests/restclient/sender.go]: Either an issue in one of the projects and resuse (not ideal), or creating an issue in each project when pushing the PR just not not lose track of it.

**Pull request merged**